src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -181,6 +181,10 @@ private Pageable mapSortParameters(Pageable pageable) {
               "The attribute exportDeclarationReference cannot be null if isExportDeclarationRequired is true."));
     }
 
+    if (bookingRequest.getVesselIMONumber() == null && bookingRequest.getExportVoyageNumber() == null && bookingRequest.getExpectedDepartureDate() == null) {
```

Comment:
You are forgetting the arrival date in this check - if the above 3 fields need to be combined with arrivalDate = null